File: app/src/main/java/com/shuyu/apprecycler/LazyApplication.java
package com.shuyu.apprecycler;

import android.app.Application;

import com.shuyu.apprecycler.chat.utils.ChatConst;

/**
 * Created by guoshuyu on 2017/9/8.
 */

public class LazyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        ChatConst.ChatInit(this);
    }

}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/LazyApplication.kt
package example.shuyu.recycler.kotlin

import android.app.Application

import example.shuyu.recycler.kotlin.chat.utils.ChatConst

/**
 * Created by guoshuyu on 2017/9/8.
 */

open class LazyApplication : Application() {

    companion object {
        @Volatile lateinit var INSTANCE: LazyApplication
    }

    override fun onCreate() {
        super.onCreate()
        ChatConst.ChatInit(this)
        INSTANCE = this
    }

}
